app/controllers/guest_rooms_controller.rb

```diff
@@ -0,0 +1,13 @@
+class GuestRoomsController < ApplicationController
+  def create
+    room = Room.find(params[:room_id])
+    gr = GuestRoom.new(room: room, guest_id: params[:id])
+
+    if gr.save
+      flash[:success] = "Room created!"
+      redirect_to "/guests/#{params[:id]}"
+    else
+      flash[:error] = "Room creation failed: #{gr.errors.full_messages.join}"
+    end
+  end
+end
```

Comment:
👍 

Great job going the extra mile here and considering the sad path. The only thing is what is the controller to do if the `else` is hit? There is no redirect or render, so the action will attempt to render a page called `create.html.erb` which doesn't exist, and then call the request to error out.